bfss3/bfss3.go

```diff
@@ -97,6 +100,14 @@ func init() {
 			return nil, err
 		}
 
+		if s := query.Get("assume_role"); s != "" {
+			cfg = aws.Config{
```

Comment:
you don't want to override the whole config, you want to just `cfg.Credentials = `, see https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials/stscreds#hdr-Assume_Role